worker-markup-core/src/main/java/com/github/cafdataprocessing/worker/markup/core/MarkupHeadersAndBody.java

```diff
@@ -35,9 +35,12 @@
     private static final Logger LOG = LoggerFactory.getLogger(MarkupHeadersAndBody.class);
 
     public static final String UNREADABLE_HEADER = "UnreadableHeader";
-    public static final String FROM_FIELD_WITH_ASTERISKS_SPLIT_ONTO_TWO_LINES_REGEX =
-            "(?<FirstPartFromField>[> ]{0,}\\*.*:\\*[A-z0-9][-A-z0-9_\\+\\.]*[A-z0-9]@[A-z0-9][-A-z0-9\\.]*[A-z0-9]\\.[A-z0-9]{1,3}[\\r]?)\\n" +
-            "(?<SecondPartFromField>[> ]{0,}\\[.*:[A-z0-9][-A-z0-9_\\+\\.]*[A-z0-9]@[A-z0-9][-A-z0-9\\.]*[A-z0-9]\\.[A-z0-9]{1,3}\\].*[\\r]?)";
+    public static final String FROM_FIELD_WITH_ASTERISKS_SPLIT_ONTO_TWO_LINES_REGEX
+        = "(?<FirstPartFromField>[> ]{0,}\\*.*:\\*[A-z0-9][-A-z0-9_\\+\\.]*[A-z0-9]@[A-z0-9][-A-z0-9\\.]*[A-z0-9]\\.[A-z0-9]{1,3}[\\r]?)?+\\n"
```

Comment:
The possessive quantifiers should make the find() end faster because it avoids backtracking repetitions. In other words, once it finds a match for one character, it continues to follow the regex pattern and does not release it; so, if it understands that it cannot match the pattern, it just gives it up, without trying other combinations (https://chortle.ccsu.edu/FiniteAutomata/Section09/sect09_21.html). 
It may change the matches in confront of the greedy and lazy quantifiers; it has not happened for what concern our integration tests. Should I remove it?

Reply:
It's not so much the possessive quantifier that I'm questioning here - rather it is the use of the `?` (i.e. 0 or 1) - presumably beforehand this would always have had to match whereas now it is optional?

Reply:
As discussed, we can't make the first line optional.